src/core/agent/agent.types.ts

```diff
@@ -49,11 +49,37 @@ type ConnectionNoteDetails = {
   message: string;
 };
 
+type IpexMessages = {
+  exn: {
+    v: string;
+    t: string;
+    d: string;
+    i: string;
+    p: string;
+    dt: string;
+    r: string;
+    q: any;
```

Comment:
Can we replace `any` with a JSON object type? https://dev.to/ankittanna/how-to-create-a-type-for-complex-json-object-in-typescript-d81 - not sure if we have this in our codebase already anywhere